exist-core/src/main/java/org/exist/xquery/PathExpr.java

```diff
@@ -116,7 +116,9 @@ public void addPredicate(final Predicate predicate) {
     public void replace(final Expression oldExpr, final Expression newExpr) {
         final int idx = steps.indexOf(oldExpr);
         if (idx < 0) {
-            LOG.warn("Expression not found: " + ExpressionDumper.dump(oldExpr) + "; in: " + ExpressionDumper.dump(this));
+            if (LOG.isTraceEnabled()) {
+                LOG.trace("Expression not found when trying to replace: " + ExpressionDumper.dump(oldExpr) + "; in: " + ExpressionDumper.dump(this));
```

Comment:
why not `LOG.trace("Expression not found when trying to replace: {}; in: {}",  ExpressionDumper.dump(oldExpr) , ExpressionDumper.dump(this));` ?